File: 2606-find-the-substring-with-maximum-cost/2606-find-the-substring-with-maximum-cost.cpp
class Solution {
public:
     int maximumCostSubstring(string s, string chars, vector<int>& vals) {
       vector<int>ch(26);  //set
        for(int i=0;i<26;i++)
        {
            ch[i]=i+1;
        }
        int cnt =0;
        for(char c:chars)   //update
        {
           ch[c-'a'] =vals[cnt++];
        }
         int maxsub=0;
         int cur=0;
         for(char c:s)
         {
             cur+=ch[c-'a'];
             if(cur<0)
             {
                 cur=0;
             }
             maxsub=max(maxsub,cur);
         }
         return maxsub;
    }
};

File: 2606-find-the-substring-with-maximum-cost/2606-find-the-substring-with-maximum-cost.py
class Solution:
   def maximumCostSubstring(self, a: str, c: str, x: List[int]) -> int:
        y=[]
        for i in a:
            if i in c: y.append(x[c.index(i)])
            else: y.append(ord(i)-96)
        s=t=0
        for i in y:
            t=max(t,0)
            s=max(s,t)
            t+=i
        return max(s,t,0)